src/server.cpp

```diff
@@ -818,9 +828,11 @@ static CTimingMeas JitterMeas ( 1000, "test2.dat" ); JitterMeas.Measure(); // TE
         else
         {
             // processing with multithreading
-// TODO optimization of the MTBlockSize value
-            const int iMTBlockSize = 10; // every 10 users a new thread is created
-            const int iNumBlocks   = ( iNumClients - 1 ) / iMTBlockSize + 1;
+            // spread work equally among available threads
+            // since decoding is a much lighter workload, only create new tasks for every 10 clients
+            // measurements by menzels suggest it is ~6% (decode) vs 60% (encode) of process cpu time
+            const int iNumBlocks   = std::min(iNumClients / 10 + 1, iMaxNumThreads);
+            const int iMTBlockSize = (iNumClients - 1) / iNumBlocks + 1; 
```

Comment:
I think this is probably not correct. If `iNumClients` is 1, then `iMTBlockSize` will be 0, and further down (not in the diff), `iStopChanCnt` will be -1, which doesn't sound right to me.

I'm not sure the reason for swapping `iNumBlocks` and `iMTBlockSize`.

Reply:
Sorry, I inferred brackets where there were none. My mistake.